sda-commons-web-autoconfigure/src/main/java/org/sdase/commons/spring/boot/web/client/SdaOidcClientConfiguration.java

```diff
@@ -9,41 +9,63 @@
 import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties.Provider;
 import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientProperties.Registration;
 import org.springframework.boot.autoconfigure.security.oauth2.client.OAuth2ClientPropertiesRegistrationAdapter;
-import org.springframework.context.annotation.Bean;
+import org.springframework.lang.Nullable;
 import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager;
 import org.springframework.security.oauth2.client.InMemoryOAuth2AuthorizedClientService;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager;
 import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService;
 import org.springframework.security.oauth2.client.registration.ClientRegistration;
-import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
 import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.stereotype.Component;
 
-@AutoConfiguration
+@Component
 @ConditionalOnProperty(value = "oidc.client.enabled", havingValue = "true")
 public class SdaOidcClientConfiguration {
 
+  private static final Logger LOG = LoggerFactory.getLogger(SdaOidcClientConfiguration.class);
+
+  /** the oidc client id */
+  @Value("${oidc.client.id}")
+  private String clientId;
+  /** the oidc client secret */
+  @Value("${oidc.client.secret}")
+  private String clientSecret;
+  /** the oidc client issuer URI */
+  @Value("${oidc.client.issuer.uri}")
+  private String oidcIssuer;
+
+  private OAuth2TokenProvider oAuth2TokenProvider;
+
   /**
-   * This bean overwrites the {@code spring.security.oauth2.client}-based autoconfigured client
-   * registrations to have more control about the initialization of the required beans for oidc
-   * client authentication.
+   * Creates and returns an instance of {@link OAuth2TokenProvider}. It calls the issuer URI to
+   * retrieve the OAuth information. In case it already happended, it will not call the endpoint
+   * again, but return the oAuth2TokenProvider right away. When it fails to gather information on
+   * the issuer endpoint it will return a null object, so it will try again in the next request
    *
-   * @param clientId the oidc client id
-   * @param clientSecret the oidc client secret
-   * @param oidcIssuer the oidc issuer url
-   * @return the client registration repository containing the oidc client registration.
+   * @return oAuth2TokenProvider
    */
-  @Bean
-  InMemoryClientRegistrationRepository clientRegistrationRepository(
-      @Value("${oidc.client.id}") String clientId,
-      @Value("${oidc.client.secret}") String clientSecret,
-      @Value("${oidc.client.issuer.uri}") String oidcIssuer) {
+  @Nullable
+  public OAuth2TokenProvider oAuth2Provider() {
```

Comment:
Seems like you implementing a manual lazy loading.

It should be possible to annotate a Bean with` @Lazy` and then use `applicationContext.getBean(OAuth2TokenProvider.class);`  in the `RequestInterceptor`to control when the bean is instantiated the first time